### PR TITLE
feat: Telegram bot

### PR DESCRIPTION
**Pull request by rsrbk**
## Description

Adds a new class `TelegramBot` that creates a bot for a specific Telegram server, which is fully powered by the underlying `ChatAgent` object.

## Motivation and Context

We want users of CAMEL framework to set up telegram and other types of bots in one click.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: camel/bots/telegram_bot.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+        # Register the message handler within the constructor
+        self.bot.message_handler(func=lambda message: True)(self.on_message)
+
+    def run(self) -> None:
+        r"""Start the Telegram bot."""
+        print("Telegram bot is running...")
+        self.bot.infinity_polling()
+
+    def on_message(self, message: 'Message') -> None:
```
Comment (Wendong-Fan) [inline camel/bots/telegram_bot.py:66]: refer directly to the expected type?
```suggestion
    def on_message(self, message: Message) -> None:
```

File: camel/bots/telegram_bot.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        Args:
+            message (types.Message): The incoming message object.
+
+        Returns:
+            None
+        """
+        self.chat_agent.reset()
+
+        user_msg = BaseMessage.make_user_message(
+            role_name="User", content=message.text
```
Comment (Wendong-Fan) [inline camel/bots/telegram_bot.py:78]: add check here in case `message.text` is empty?

File: camel/bots/telegram_bot.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        self.bot.infinity_polling()
+
+    def on_message(self, message: 'Message') -> None:
+        r"""Handles incoming messages from the user.
+
+        Args:
+            message (types.Message): The incoming message object.
+
+        Returns:
+            None
```
Comment (Wendong-Fan) [inline camel/bots/telegram_bot.py:73]:
```suggestion
```

**Review by Wendong-Fan — COMMENTED** [review]
Thanks for the contribution! Left some comments

**Review by Wendong-Fan — APPROVED** [review]

**Review by willshang76 — APPROVED** [review]

**Pull request merged**